### PR TITLE
Issue #475 - implement support for conformsTo function

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: fhir-path/src/main/java/com/ibm/fhir/path/function/ConformsToFunction.java [hunk 1]
```diff
@@ -43,20 +55,36 @@ public int getMaxArity() {
...
+            
+            FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();    
+            for (Constraint constraint : ProfileSupport.getConstraints(url, modelClass)) {
+                try {
+                    Collection<FHIRPathNode> result = evaluator.evaluate(evaluationContext, constraint.expression(), context);
+                    if (evaluatesToBoolean(result) && isFalse(result)) {
+                        return SINGLETON_FALSE;
+                    }
+                } catch (FHIRPathException e) {
+                    // TODO: log this
```
Comment (lmsurpre) [inline fhir-path/src/main/java/com/ibm/fhir/path/function/ConformsToFunction.java]: we have https://github.com/IBM/FHIR/issues/187 for cleaning up TODOs, but obviously better just to add now it if its quick/easy

File: fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java [hunk 2]
```diff
@@ -36,6 +36,28 @@ public static FHIRRegistry getInstance() {
         return INSTANCE;
     }
     
+    public boolean hasResource(String url) {
+        if (Objects.isNull(url)) {
+            return false;
+        }
+        
+//      String id = null;
```
Comment (lmsurpre) [inline fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java]: why is this commented out?  is it needed?

File: fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java [hunk 3]
```diff
@@ -36,6 +36,28 @@ public static FHIRRegistry getInstance() {
...
+        int index = url.indexOf("#");
+        if (index != -1) {
+//          id = url.substring(index + 1);
```
Comment (lmsurpre) [inline fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java]: why is this commented out?  is it needed?

File: fhir-path/src/main/java/com/ibm/fhir/path/function/ConformsToFunction.java (hunk 1 above)
Comment (JohnTimm) [reply fhir-path/src/main/java/com/ibm/fhir/path/function/ConformsToFunction.java]: added logging to the catch block

File: fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java (hunk 2 above)
Comment (JohnTimm) [reply fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java]: removed

File: fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java (hunk 3 above)
Comment (JohnTimm) [reply fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java]: removed

**Pull request merged**